### PR TITLE
Add capability for `ControllerPublishVolume.readonly` field

### PR DESCRIPTION
**Pull request by saad-ali**
Closes https://github.com/container-storage-interface/spec/issues/293

### CODE REVIEW

**Comment by saad-ali** [comment]
Will change this per https://github.com/container-storage-interface/spec/issues/293#issuecomment-437195734

**Comment by saad-ali** [comment]
@julian-hj @jieyu per our discussion, I changed this from a comment to a new capability.

File: csi.proto [hunk 1]
```diff
@@ -576,8 +576,9 @@ message ControllerPublishVolumeRequest {
   // This is a REQUIRED field.
   VolumeCapability volume_capability = 3;
 
-  // Whether to publish the volume in readonly mode. This field is
-  // REQUIRED.
+  // Whether to publish the volume in readonly mode. This field MUST be
+  // specified only if SP has PUBLISH_READONLY controller capability.
+  // This is a REQUIRED field.
   bool readonly = 4;
```
Comment (gnufied) [inline csi.proto]: Is making `readonly` flag optional simpler?

Comment (saad-ali) [reply csi.proto]: Problem with making it optional is that the CO doesn't know when to set it. If CO sets it to true and a plugin doesn't support it, what happens? Consider the case where the CO uses the same flag for both ControllerPublish and NodePublish readonly -- we don't want to get in to a case where the CO thinks a volume is readonly but it's not.

File: csi.proto [hunk 2]
```diff
@@ -576,8 +576,9 @@ message ControllerPublishVolumeRequest {
   // This is a REQUIRED field.
   VolumeCapability volume_capability = 3;
 
-  // Whether to publish the volume in readonly mode. This field is
-  // REQUIRED.
+  // Whether to publish the volume in readonly mode. This field MUST be
+  // specified only if SP has PUBLISH_READONLY controller capability.
```
Comment (jieyu) [inline csi.proto]: `This field MUST be REQUIRED specified only if` conflicts with `This is a REQUIRED field`.

We should say that if the SP does not have PUBLISH_READONLY capability, the CO will set this field to `false`.

**Comment by saad-ali** [comment]
Feedback addressed PTAL

**Pull request merged**